experiments/fishyscapes.py

```diff
@@ -248,6 +250,129 @@ def wrapper(image):
     _run.info['awesomemango_anomaly2'] = fs.evaluate(wrapper, data)
 
 
+@ex.command
+def ood_ratio(testing_dataset, _run, _log, validation=False):
+    # BELOW, IMPORT ANY OF YOUR NETWORK CODE
+    from deeplabv3 import DeepWV3PlusTH
+
+    model = DeepWV3PlusTH(num_classes=19).cuda()
+    model.load_state_dict(torch.load('experiments/model_7.pth'))
+    model.eval()
+
+    fsdata = FSData(**testing_dataset)
+
+    # BELOW CODE IS NECESSARY FROM US
+    # Hacks because tf.data is shit and we need to translate the dict keys
+    def data_generator():
+        dataset = fsdata.validation_set if validation else fsdata.testset
+        for item in dataset:
+            data = fsdata._get_data(training_format=False, **item)
+            out = {}
+            for m in fsdata.modalities:
+                blob = crop_multiple(data[m])
+                if m == 'rgb':
+                    m = 'image_left'
+                if 'mask' not in fsdata.modalities and m == 'labels':
+                    m = 'mask'
+                out[m] = blob
+            yield out
+
+    data_types = {}
+    for key, item in fsdata.get_data_description()[0].items():
+        if key == 'rgb':
+            key = 'image_left'
+        if 'mask' not in fsdata.modalities and key == 'labels':
+            key = 'mask'
+        data_types[key] = item
+
+    data = tf.data.Dataset.from_generator(data_generator, data_types)
+
+    fs = bdlb.load(benchmark="fishyscapes", download_and_prepare=False)
+
+    # FILL IN THE WRAPPER FUNCTION TO DO A SINGLE-FRAME PREDICTION
+    def wrapper(image):
+        image = image.numpy().astype('uint8')
+        with torch.no_grad():
+            img = torch.from_numpy(image).float().unsqueeze(0).permute(0, 3, 1, 2) / 255.
+            # img should be in [0, 1] and of shape 1x3xHxW
+            logit, logit_ood = model(img.cuda())
+        out = torch.nn.functional.softmax(logit_ood, dim=1)
+        p1 = torch.logsumexp(logit, dim=1) # ln hat_p(x|din)
+        p2 = out[:, 1]  # p(~din|x)
+        probs = (- p1) + p2.log() # - ln hat_p(x|din) + ln p(~din|x)
+        probs = probs[0].cpu()
+        # output is HxW
+        return probs
+
+    _run.info['ood_ratio_rev1'] = fs.evaluate(wrapper, data)
+
+
+@ex.command
+def nflowjs(testing_dataset, _run, _log, validation=False):
+    # BELOW, IMPORT ANY OF YOUR NETWORK CODE
+    from deeplabv3 import DeepWV3PlusTH
+
+    model = DeepWV3Plus(num_classes=19).cuda()
```

Comment:
I cannot import this model, and if I change this to `DeepWV3PlusTH`, some weights are missing in the model you submitted.

Reply:
Sorry, fixed it with the latest commit.